CIPs/cip-124.md

```diff
@@ -71,195 +60,113 @@ By gossiping about ranges of streams ceramic nodes that are arbitrarily out of s
-```tsx
-// Proposed EventID
-<EventID> ::=
-  <multibase-prefix 'k' base36>
-  <multicodec 0xce streamid>
-  <stream-type varint>
-  <inline-cid>
-
-<inline-cid> ::=
-  <cidv1 0x01>
-  <multicodec-content-type 0x71 dag-cbor >
-  <hash-function-code 0x00 identity>
-  <dag-cbor-size-in-bytes varint >
-  <dag-cbor-bytes [
-    height: number, 
-    event-cid: link
-  ]>
-```
-[CIDv1](https://github.com/multiformats/cid)
-```
-<cidv1> ::=
-  <multibase-prefix>
-  <multicodec-cidv1>
-  <multicodec-content-type>
-  <multihash-content-address>
-```
-[Multihash](https://github.com/multiformats/multihash)
-```
-<multihash-content-address> ::=
-  <varint hash function code>
-  <varint digest size in bytes>
-  <hash function output>
-```
-[multiformats/unsigned-varint](https://github.com/multiformats/unsigned-varint)
-
-### **Ceramic node discovery and interest advertisements**
-In order to discover peers with over lapping interests nodes must advertise there interest ranges to the network.
-
-#### **Bootstrap**
-In order to allow the Ceramic nodes to find each other we have a small set of bootstrap nodes with known locations.
-When a node comes up it will call out the the bootstrap node to initialize the list of neighbors.
-This list should be override-able by the environment for testing or closed network.
-
-#### **Neighbor gossip**
-Once a node has other nodes in its neighbors list it should periodically call each neighbor
-and exchange neighbors list. Over time each node should learn about all other ceramic nodes. 
-
-```ts
-struct Neighbor {
-    location: [multiaddr]
-    last_seen: [us_since_epoch]
-    latency: [ms_ping_rtt]
-}
-
-array Neighbors [Neighbor]
-```
-
-A node should probably limit the nodes it responds with to nodes that have be seen recently. e.g. Last 28 days.
-When a new node comes up its existence will get gossiped around the network.
-
-#### **Interests Gossip**
-The Interests are ranges over the recon key space.
-```
-(start, (ahash, cid?), divider, (ahash, cid?), stop)
-```
-
-The messages are sent as CBOR
-```ts
-struct HaveMessage {
-    k: [String], // keys must be 1 longer then hashes unless both are empty
-    h: [String], // hashes must be 1 shorter then keys
-    cid?: [CID], // optional CID of the sub-tree node if remote is a static blob store
-}
-
-struct Interests [
-  {
-    start: String,
-    stop: String,
-    have: HaveMessage,
-  }
-]
+### Stream Set
+
+A stream set is defined by two values, `separator-key` and `separator-value`. A stream is considered to be part of a stream set its InitEvent contains a header field that maps to the *separator-key* and the value of this field is equal to the *separator-value*. For example, stream set containing all *Model Instance Documents* in the *Model* stream `kjzl6hvfrbw6c82mkud4qs38zl4hd03ifoyg2ksvfjkhuxebfzh3ef89vwvtvrr` would be defined as:
```

Comment:
Ok, I think to start we mainly want to support the sep-key/value sorting? I mainly wanted to call those values out so that they could be used for the eventIds below. Do you think that we should rename this section maybe?